Mathlib/Algebra/BigOperators/Finprod.lean

```diff
@@ -920,6 +921,14 @@ theorem finprod_mem_sUnion {t : Set (Set α)} (h : t.PairwiseDisjoint id) (ht₀
   rw [Set.sUnion_eq_biUnion]
   exact finprod_mem_biUnion h ht₀ ht₁
 
+@[to_additive]
+lemma finprod_option [Finite α] (φ : Option α → M) : ∏ᶠ o, φ o = φ none * ∏ᶠ a, φ (some a) := by
```

Comment:
This should only need that `(mulSupport φ).Finite`, right?

Reply:
Isn't `(mulSupport φ).Finite` equivalent and more natural?

Reply:
Composition with `some` feels more likely to reduce to something useful; consider when `φ = Option.map f`